graylog2-server/src/main/java/org/graylog2/alarmcallbacks/AlarmCallbackConfigurationServiceMJImpl.java

```diff
@@ -94,4 +100,13 @@ private AlarmCallbackConfigurationAVImpl implOrFail(AlarmCallbackConfiguration c
             throw new IllegalArgumentException("Supplied output must be of implementation type AlarmCallbackConfigurationAVImpl, not " + callback.getClass());
         }
     }
+
+    // Remove duplicated ID stored in `id` field to avoid MongoJack fetching it instead of `_id`.
+    // See https://github.com/Graylog2/graylog2-server/issues/1428 for more details.
+    public void migrate() {
```

Comment:
I saw we did something like this on `MongoIndexRangeService` and thought it was a bit cleaner than having all the logic in the periodical, but you are probably right. Will take care of it.